File: lib/commonAPI/jxcore/ext/platform/android/src/com/rho/jxcore/JxcoreSingleton.java
package com.rho.jxcore;

import java.util.LinkedList;
import java.util.List;

import com.rhomobile.rhodes.api.IMethodResult;

class JxcoreSingleton extends JxcoreSingletonBase implements IJxcoreSingleton {
    public JxcoreSingleton(JxcoreFactory factory) {
        super(factory);
    }

    List<Object> getIDs() {
        List<Object> ids = new LinkedList<Object>();
        ids.add("SCN1");
        ids.add("SCN2");
        return ids;
    }
    
    @Override
    protected String getInitialDefaultID() {
        return (String)(getIDs().get(0));
    }

    @Override
    public void enumerate(IMethodResult res) {
        res.set(getIDs());
    }
}

File: lib/commonAPI/jxcore/ext/platform/iphone/impl/JxcoreSingleton.m

#import "JxcoreSingleton.h"


@implementation JxcoreSingleton


-(NSString*)getInitialDefaultID {
    return @"SCN1";
}


-(void) enumerate:(id<IMethodResult>)methodResult {
    NSMutableArray* ar = [NSMutableArray arrayWithCapacity:2];
    [ar addObject:@"SCN1"];
    [ar addObject:@"SCN2"];
    [methodResult setResult:ar];
}




@end